### ISSUE TITLE
kdb-import `import` Strategy Doesn't Work as Described

### ISSUE THREAD

**Issue by iandonnelly** [ElektraInitiative/libelektra#329]
The `import` strategies doesn't work as described in the description of the `-s` option. The description is:

```
-s --strategy <name>     Specify which strategy should be used to resolve conflicts.
                     More precisely, strategies are used to handle deviations from the
                     base version of a key.
                     When and which strategies are used and what they do depends
                     mostly on the used base KeySet.

                     Note: For a two-way merge, the `ours` version of the keys is used
                     in place of `base`

                     Currently the following strategies exist
                       preserve      .. automerge only those keys where just one
                                        side deviates from base (default)
                       ours          .. like preserve, but in case of conflict use our version
                       theirs        .. like preserve, but in case of conflict use their version
                       cut           .. primarily used for import. removes existing keys below
                                        the import point and always takes the imported version
                       import        .. primarily used for import. preserves existing keys if
                                        they do not exist in the imported keyset. in all other
                                        cases the imported keys have precedence
```

Yet when using this option:

```
$ kdb import -s import user/ian/ini ini < ../tests/import.ini
2 conflicts were detected that could not be resolved automatically:  
user/ian/ini/key2
ours: CONFLICT_SAME, theirs: CONFLICT_DELETE

user/ian/ini/key4
ours: CONFLICT_SAME, theirs: CONFLICT_DELETE

Merge unsuccessful.
```

According to the description, the keys should have been imported successfully replacing the current version.

Either the description needs to be changed or this `-s import` option should work as currently described (the optimal solution).

**Comment by markus2330** [comment]
Thanks for reporting! We should add testcases for the import strategies on `kdb` level, too.

**Comment by markus2330** [comment]
Is this issue still open?
@iandonnelly can you please post import.ini and the content of `user/ian/ini`?

**Comment by fberlakovich** [comment]
The documentation is definitely confusing. What the import strategy actually does is using the OneSide***Value***Strategy. This means that the value of imported keys always have precedence. However, as you have already noticed this strategy will refuse to merge a case where a key is present only in the imported keyset. 

This can be easily changed, however, I am not sure what the desired behaviour is. When I implemented these strategies I used check_import.sh as a guideline and tried to mimic the previous implementation.

**Comment by markus2330** [comment]
I am afraid that there are too many merging strategies without any real reason for some of them.  Reimplementing old behaviour was not really necessary. It is also very confusing that the class-names of the strategies are named differently then the command-line switches.
Finally, the docu is now also cluttered around everywhere:
- doc/help/IMPORT.md
- doc/help/MERGE.md
- src/tools/kdb/cmdline.cpp
- doc/tutorials/merge.md

So we need:
- a clever way to have a structure in the strategies name (that is the same as the class-names) which makes clear what the strategy will do. Things like "import" strategy should only be aliases (if needed at all).
- and we should have a single document describing all strategies, tutorials referring to it

In conclusion: Its very nice that we have such a flexible way to combine all strategies (@fberlakovich really nice done) and we also have a nice high-level strategy layer, but still missing is a 1:1 mapping of the high-level merging strategies and the strategies for command-line interfaces as well as a proper documentation of these concepts.

**Comment by markus2330** [comment]
As written by @sanssecours the tutorial needs to be improved, too:

"I think we should add kdb commands to the tutorial that generate the keysets system/base, system/ours, system/base.

The description of the strategies makes sense, when we only look at key4. However I do not understand how the strategies determine some of the other keys in the merged keyset. For example, why do we lose key2 if we use the strategy ours?"
